### ISSUE TITLE
Easy way to prevent multiple instances of the same event listener to be attached to the event

### ISSUE THREAD

**Issue by sanjaypojo** [voximplant/react-native-voximplant#74]
We have a fairly complex event lifecycle, and I believe that our event listener function `Voximplant.ClientEvents.IncomingCall` is getting bound twice as the event is being emitted twice at our end for an incoming call. Is there a way for me to check if any event handler already exists, before adding one? If I use `client.off`, I have no way of checking that it worked correctly. Hope this makes sense and please let me know if you need any other info. I'm using `v1.6.0`

**Comment by YuliaGrigorieva** [comment]
Hello!

It seems strange as we use Set to store event listeners: https://github.com/voximplant/react-native-voximplant/blob/b8e40a283c3903fc233f9b56e7d2310a1443a9a7/src/client/Client.js#L110

So even an event listener is registered twice, Set should prevent emitting an event more than once on the same event listener.

To investigate this issue, I suggest that you debug the JS code with the breakpoint on this line for `Voximplant.ClientEvents.IncomingCall` event: https://github.com/voximplant/react-native-voximplant/blob/b8e40a283c3903fc233f9b56e7d2310a1443a9a7/src/client/Client.js#L476
and check the event listeners that are set for `Voximplant.ClientEvents.IncomingCall`:
<img width="1682" alt="Screenshot 2019-05-06 at 18 04 31" src="https://user-images.githubusercontent.com/8545374/57235118-d3735880-702a-11e9-9e81-e718795fe349.png">

If you can reproduce the issue on the demo app or can provide the steps / SDK API method call sequence, please let me know, so I can reproduce it.

Regarding your question about checking currently registered event listeners for an event, we do not have this functionality right now, but we will consider your feature request.

Please let me know the results of the debugging.

Best regards,
Yulia Grigorieva

**Comment by sanjaypojo** [comment]
Thanks Yulia! Looks like this is not due to a duplicate listener, it is just that voximplant fires multiple `IncomingCall` events while we are processing the call. We perform some operations before accepting the call -- maybe voximplant retries the call events while it is waiting?

**Comment by YuliaGrigorieva** [comment]
Hello @sanjaypojo ,

Voximplant React Native SDK fires only one IncomingCall event for each incoming call. 

Could you please check if the call IDs in the `IncomingCall` events are the same or not?
If the call IDs are the same, it means that the same call is processed several times. If they are different, it means that the client receives different calls. 

If you can reproduce the issue, please collect the Voximplant React Native SDK logs and send them to us at support@voximplant.com for further investigation.

Best regards,
Yulia Grigorieva

**Comment by sanjaypojo** [comment]
We are investigating and will get back to you with logs. In the meantime, I have a related question. If I declare the listener using an anonymous function, then how do I remove it? For example:
```javascript
call.on(Voximplant.CallEvents.Disconnected, (event) => {
  console.log(event)
}
```

**Comment by YuliaGrigorieva** [comment]
Hello @sanjaypojo !

This may be the issue. I have taken this case into account and will implement the ability to remove anonymous event handler for the next release.

Best regards,
Yulia Grigorieva

**Comment by sanjaypojo** [comment]
That would be great, thanks @YuliaGrigorieva

**Comment by YuliaGrigorieva** [comment]
Hello @sanjaypojo ,

We have just released Voximplant React Native SDK 1.8.0 that contains the fix for this issue.

Please use `client.off(Voximplant.ClientEvents.IncomingCall)` to remove all anonymous functions that handle incoming call event.
You can use the same construction to remove event handlers in other APIs such as Call, AudioDeviceManager etc

Best regards,
Yulia Grigorieva
